### DISCUSSION TITLE
Potential Additions

### DISCUSSION THREAD

**Discussion by coree** [coree/awesome-rag#1] [Ideas]
Welcome to the **Potential Additions** discussion!   
Everyone is encouraged to suggest new entries. Please format your suggestions using Markdown like this: 
```- [ ] `[Category]` Your Suggestion```
 where `Category` should be either `Paper`, `Tool`, `Resource`, or `Collection`. The checkbox will help us track whether an addition has been included.

**Comment by coree** [comment]
- [ ] `[Paper]` Hsia, Jennifer, et al. "RAGGED: Towards Informed Design of Retrieval Augmented Generation Systems." arXiv preprint arXiv:2403.09040 (2024).

**Comment by coree** [comment]
-  [ ] `[Paper]` Liu, Yanming, et al. "Ra-isf: Learning to answer and understand from retrieval augmentation via iterative self-feedback." arXiv preprint arXiv:2403.06840 (2024).

**Comment by coree** [comment]
- [ ] `[Resource]` DeepLearning.AI course [Knowledge Graphs for RAG](https://www.deeplearning.ai/short-courses/knowledge-graphs-rag)

**Comment by coree** [comment]
- [ ] `[Resource]` Blog post on PromptingGuide.AI on [RAG for LLMs](https://www.promptingguide.ai/research/rag)

**Comment by coree** [comment]
- [ ] `[Paper]` Xu, Shicheng, et al. "Unsupervised Information Refinement Training of Large Language Models for Retrieval-Augmented Generation." arXiv preprint arXiv:2402.18150 (2024).

**Comment by jmulal** [comment]
- [x] `[Paper]`Asai, A., Zhong, Z., Chen, D., Koh, P.W., Zettlemoyer, L., Hajishirzi, H. and Yih, W.T., 2024. Reliable, Adaptable, and Attributable Language Models with Retrieval. arXiv:2403.03187.

**Comment by coree** [comment]
- [ ] `[Tool]` Cohere [Command-R](https://txt.cohere.com/command-r/) for production ready RAG

**Comment by coree** [comment]
- [ ] `[Paper]` Liao, Huanxuan et al. “Imagination Augmented Generation: Learning to Imagine Richer Context for Question Answering over Large Language Models.” (2024)

**Comment by VityaVitalich** [comment]
- [ ] `[Paper]` Moskvoretskii, Viktor, et al. "Adaptive Retrieval Without Self-Knowledge? Bringing Uncertainty Back Home." arXiv preprint arXiv:2501.12835 (2025).

**Comment by jeffreyflynt** [comment]
[Tool] Synthetic corporate dataset generator for AI agent evaluation. [OrgForge](https://github.com/aeriesec/orgforge)

**Comment by MuathJ** [comment]
- [ ] `[Tool]` [LocalMode](https://github.com/LocalMode-AI/LocalMode) - Local-first, privacy-first RAG framework that runs entirely in the browser: embeddings, a vector DB with HNSW index, semantic cache, and agents. No servers, no API keys.

**Comment by Officially-aditya** [comment]
- [ ] `[Resource]` [RAG + Knowledge Graph Master Course](https://www.thequery.in/books/rag-kg-master-course) by Addy. A free, 16-chapter written book covering retrieval, knowledge graphs, hybrid systems, and production projects. Available as HTML and EPUB without registration.

Disclosure: I maintain TheQuery and am suggesting my own resource. This may fit the Resources section as a longer written learning resource alongside the existing lectures and tutorials.

**Comment by aifabrice** [comment]
- [ ] `[Tool]` [Jev RAG](https://github.com/aifabrice/jev-rag) — Open-source local document search and RAG with a vector-free SQLite FTS5/BM25 default, Jev evidence reranking, optional BM25 + embedding hybrid retrieval, cited streaming answers, and a reproducible BEIR NFCorpus benchmark.

Disclosure: I maintain this project.

**Reply by aifabrice** [reply]
Update: the interactive benchmark is now live at https://aifabrice.github.io/jev-rag/

It compares seven pipelines on the complete BEIR NFCorpus test split (3,633 documents / 323 queries), from BM25 through agentic lexical search and optional hybrid retrieval, with and without Jev reranking. The page includes the recorded metrics and links to the exact configuration and caveats. These are project-run results, not an official MTEB submission.
